integrations/octopus/client.py

```diff
@@ -0,0 +1,102 @@
+        self,
+        endpoint: str,
+        params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+        method: str = "GET",
+    ) -> Any:
+        """Send a request to the Octopus Deploy API."""
+        url = f"{self.octopus_url}{endpoint}"
+        response = await self.client.request(
+            url=url,
+            method=method,
+            headers=self.api_auth_header,
+            params=params,
+            json=json_data,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPStatusError as e:
+            logger.error(
+                f"Got HTTP error to url: {url} with status code: {e.response.status_code} and response text: {e.response.text}"
+            )
+            raise
+        return response.json()
+
+    async def get_paginated_resources(
+        self,
+        kind: str,
+        params: Optional[dict[str, Any]] = None,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        """Fetch paginated data from the Octopus Deploy API."""
+        if params is None:
+            params = {}
+        params["skip"] = 0
+        params["take"] = PAGE_SIZE
+        page = 0
+        while True:
+            response = await self._send_api_request(f"{kind}s", params=params)
+            items = response.get("Items", [])
+            last_page = response.get("LastPageNumber", 0)
+            yield items
+            if page >= last_page:
+                break
+            params["skip"] += PAGE_SIZE
+            page += 1
+
+    async def get_single_entity(
+        self, entity_kind: str, entity_id: str
+    ) -> dict[str, Any]:
+        """Get a single entity by kind and ID."""
+        return await self._send_api_request(f"{entity_kind}/{entity_id}")
+
+    async def get_all_spaces(self) -> list[dict[str, Any]]:
+        """Get all spaces in the Octopus instance."""
+        return await self._send_api_request("spaces/all")
+
+    async def _create_subscription(self, space_id: str, app_host: str) -> None:
+        """Create a new subscription for a space."""
+        endpoint = "subscriptions"
+        subscription_data = {
+            "EventNotificationSubscription": {
+                "WebhookURI": f"{app_host}/integration/webhook",
+                "WebhookTimeout": "00:00:50",
+            },
+            "IsDisabled": False,
+            "Name": "Port Subscription",
+            "SpaceId": f"{space_id}",
+        }
+        logger.info(
+            f"Creating Webhook Subscription - '{subscription_data['Name']}' in '{space_id}'"
+        )
+        return await self._send_api_request(
+            endpoint, json_data=subscription_data, method="POST"
+        )
+
+    async def create_webhook_subscription(self, app_host: str) -> dict[str, Any]:
+        """Create a new subscription."""
+        for space in await self.get_all_spaces():
+            await self._create_subscription(space["Id"], app_host)
+        return {"ok": True}
```

Comment:
instead of manually setting ok to True, can we rely on the response from the `_create_subscription`. And maybe keep track of which spaces we were able to create subscription for and which one we didn't?